sdk/core/azure-core/inc/azure/core/http/curl/curl connection_pool.hpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief The curl connection pool provides the utilities for creating a new curl connection and to
+ * keep a pool of connections to be re-used.
+ */
+
+#pragma once
+
+#include "azure/core/http/curl/curl_connection.hpp"
+#include "azure/core/http/http.hpp"
+
+#include <curl/curl.h>
+#include <list>
+#include <map>
+#include <memory>
+#include <mutex>
+
+#ifdef TESTING_BUILD
+// Define the class name that reads from ConnectionPool private members
+namespace Azure { namespace Core { namespace Test {
+  class TransportAdapter_connectionPoolTest_Test;
+}}} // namespace Azure::Core::Test
+#endif
+
+namespace Azure { namespace Core { namespace Http {
+
+  /**
+   * @brief CURL HTTP connection pool makes it possible to re-use one curl connection to perform
+   * more than one request. Use this component when connections are not re-used by default.
+   *
+   * This pool offers static methods and it is allocated statically. There can be only one
+   * connection pool per application.
+   */
+  struct CurlConnectionPool
+  {
+#ifdef TESTING_BUILD
+    // Give access to private to this tests class
+    friend class Azure::Core::Test::TransportAdapter_connectionPoolTest_Test;
+#endif
+
+    /**
+     * @brief Mutex for accessing connection pool for thread-safe reading and writing.
+     */
+    static std::mutex s_connectionPoolMutex;
```

Comment:
This is a public member.
```suggestion
    static std::mutex ConnectionPoolMutex;
```

Reply:
`s` is for static